pkg/dockerregistry/server/auth.go

```diff
@@ -167,16 +153,57 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 					challenge.err = err
 					return nil, challenge
 				}
-
-				return ctx, nil
```

Comment:
does it cause problems if we do it in all cases? not sure what to do if a mix of repository/push and admin/prune access records are requested